### PR TITLE
Added upcoming command

### PR DESCRIPTION
**Pull request by Varstak**
Added Upcoming functionality.

### CODE REVIEW

File: src/main/java/seedu/ultron/logic/parser/UltronParser.java [hunk 1]
```diff
@@ -6,16 +6,7 @@
...
-import seedu.ultron.logic.commands.ClearCommand;
-import seedu.ultron.logic.commands.Command;
-import seedu.ultron.logic.commands.DeleteCommand;
-import seedu.ultron.logic.commands.EditCommand;
-import seedu.ultron.logic.commands.ExitCommand;
-import seedu.ultron.logic.commands.FindCommand;
-import seedu.ultron.logic.commands.HelpCommand;
-import seedu.ultron.logic.commands.ListCommand;
-import seedu.ultron.logic.commands.RemarkCommand;
+import seedu.ultron.logic.commands.*;
```
Comment (Lanreath) [inline src/main/java/seedu/ultron/logic/parser/UltronParser.java:9]: Show all used imports

File: src/main/java/seedu/ultron/logic/parser/UpcomingCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class UpcomingCommandParser implements Parser<UpcomingCommand> {
+    public UpcomingCommand parse(String args) throws ParseException {
+        try {
+            String trimmedArgs = args.trim();
+            int days = ParserUtil.parseDays(trimmedArgs);
+            return new UpcomingCommand(new OpeningsBeforeDaysPredicate(days));
+        } catch (ParseException pe) {
```
Comment (Lanreath) [inline src/main/java/seedu/ultron/logic/parser/UpcomingCommandParser.java:18]: No need to catch the ParseException, the parse method already throws the exception.

File: src/main/java/seedu/ultron/logic/parser/UpcomingCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, UpcomingCommand.MESSAGE_USAGE), pe);
+        }
+    }
+
+    public static LocalDate getTime(String str) {
```
Comment (Lanreath) [inline src/main/java/seedu/ultron/logic/parser/UpcomingCommandParser.java:24]: This method belongs in ParserUtil

File: src/main/java/seedu/ultron/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -129,4 +129,13 @@ public static Set<Date> parseDates(Collection<String> dates) throws ParseExcepti
         }
         return dateSet;
     }
+
+    public static int parseDays(String days) throws ParseException {
```
Comment (Lanreath) [inline src/main/java/seedu/ultron/logic/parser/ParserUtil.java:133]: parseDay would be clearer given that the argument only takes in a single number

**Review by Lanreath — CHANGES_REQUESTED** [review]
A few nitpicks, don't forget to include documentation for the new command, parser and predicate.
Once the date comparator is done, this will simplify the date comparison in the predicate and remove the need for getTime.

File: src/main/java/seedu/ultron/logic/parser/UpcomingCommandParser.java (hunk 2 above)
Comment (Varstak) [reply src/main/java/seedu/ultron/logic/parser/UpcomingCommandParser.java:18]: When implementing parseDays, I referred to the parseIndex function which also throws exception for unsigned non-zero integers.

**Pull request merged**